neo/SmartContract/Manifest/ContractPermission.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Linq;
+
+namespace Neo.SmartContract.Manifest
+{
+    /// <summary>
+    /// The permissions field is an array containing a set of Permission objects. It describes which contracts may be invoked and which methods are called.
+    /// </summary>
+    public class ContractPermission
+    {
+        /// <summary>
+        /// The contract field indicates the contract to be invoked. It can be a hash of a contract, a public key of a group, or a wildcard *.
+        /// If it specifies a hash of a contract, then the contract will be invoked; If it specifies a public key of a group, then any contract in this group will be invoked; If it specifies a wildcard*, then any contract will be invoked.
+        /// </summary>
+        public ContractPermissionDescriptor Contract { get; set; }
+
+        /// <summary>
+        /// The methods field is an array containing a set of methods to be called. It can also be assigned with a wildcard *. If it is a wildcard *, then it means that any method can be called.
+        /// If a contract invokes a contract or method that is not declared in the manifest at runtime, the invocation will fail.
+        /// </summary>
+        public WildCardContainer<string> Methods { get; set; }
+
+        public static readonly ContractPermission DefaultPermission = new ContractPermission
+        {
+            Contract = ContractPermissionDescriptor.CreateWildcard(),
+            Methods = WildCardContainer<string>.CreateWildcard()
+        };
+
+        /// <summary>
+        /// Parse ContractPermission from json
+        /// </summary>
+        /// <param name="json">Json</param>
+        /// <returns>Return ContractPermission</returns>
+        public static ContractPermission FromJson(JObject json)
+        {
+            return new ContractPermission
+            {
+                Contract = ContractPermissionDescriptor.FromJson(json["contract"]),
+                Methods = WildCardContainer<string>.FromJson(json["methods"], u => u.AsString()),
+            };
+        }
+
+        /// <summary
+        /// To json
+        /// </summary>
+        public JObject ToJson()
+        {
+            var json = new JObject();
+            json["contract"] = Contract.ToJson();
+            json["methods"] = Methods.ToJson();
+            return json;
+        }
+
+        /// <summary>
+        /// Return true if is allowed
+        /// </summary>
+        /// <param name="manifest">The manifest of which contract we are calling</param>
+        /// <param name="method">Method</param>
+        /// <returns>Return true or false</returns>
+        public bool IsAllowed(ContractManifest manifest, string method)
+        {
+            if (Contract.IsHash)
+            {
+                if (!Contract.Hash.Equals(manifest.Hash)) return false;
+            }
+            else if (Contract.IsGroup)
+            {
+                if (manifest.Groups.All(p => !p.PubKey.Equals(Contract.Group))) return false;
```

Comment:
Does All work as expected or Sum would also work?

Reply:
What sum?

Reply:
It is checking, for all groups, if any of them is not equal.

Maybe this is strange, because one of yours groups may not allow you to enter, but another you can allow you.

Reply:
`All` not `Any`.

Reply:
If `p.PubKey.Equals(Contract.Group)` is false for any of the groups of have it will probably return false, no?

Reply:
All

Reply:
If `p.PubKey.Equals(Contract.Group)` is false for **ALL** of the groups, it will return false.